In [1]:
import numpy as np
import pandas as pd

In [158]:
sale_info = pd.read_csv('../../data/raw/EXTR_RPSale.csv')
sale_info.head(2)

ExciseTaxNbr   Major  Minor DocumentDate  SalePrice    RecordingNbr Volume  \
0       2857854  198920   1430   03/28/2017          0  20170410000541          
1       2743355  638580    110   07/14/2015     190000  20150715002686          

  Page PlatNbr PlatType  ... PropertyType PrincipalUse SaleInstrument  \
0                        ...            3            7             15   
1                        ...            3            6              3   

  AFForestLand  AFCurrentUseLand  AFNonProfitUse  AFHistoricProperty  \
0            N                 N               N                   N   
1            N                 N               N                   N   

  SaleReason PropertyClass SaleWarning  
0         16             2       20 31  
1          1             8          15  

[2 rows x 24 columns]

In [159]:
sale_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 351067 entries, 0 to 351066
Data columns (total 24 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   ExciseTaxNbr        351067 non-null  int64 
 1   Major               351067 non-null  int64 
 2   Minor               351067 non-null  int64 
 3   DocumentDate        351067 non-null  object
 4   SalePrice           351067 non-null  int64 
 5   RecordingNbr        351067 non-null  object
 6   Volume              351067 non-null  object
 7   Page                351067 non-null  object
 8   PlatNbr             351067 non-null  object
 9   PlatType            351067 non-null  object
 10  PlatLot             351067 non-null  object
 11  PlatBlock           351067 non-null  object
 12  SellerName          351067 non-null  object
 13  BuyerName           351067 non-null  object
 14  PropertyType        351067 non-null  int64 
 15  PrincipalUse        351067 non-null  int64 
 16  Sa

In [160]:
sale_info = sale_info[['Major', 'Minor', 'SalePrice', 'DocumentDate','PrincipalUse']]
sale_info.head()

Major  Minor  SalePrice DocumentDate  PrincipalUse
0  198920   1430          0   03/28/2017             7
1  638580    110     190000   07/14/2015             6
2  919715    200     192000   07/08/2019             2
3  894677    240     818161   12/21/2016             6
4  445872    260          0   10/03/2016             2

In [161]:
#Only using data related to residential properties. According to the LookUp file, '6' represents a represential property.
sale_info = sale_info[sale_info['PrincipalUse'] == 6] #filtering out
sale_info.drop(['PrincipalUse'], axis=1, inplace=True) #removing the column because now that the data contains only
#residential properties, we have no use for it to be labeled.
sale_info.head()

Major  Minor  SalePrice DocumentDate
1  638580    110     190000   07/14/2015
3  894677    240     818161   12/21/2016
5  408330   4150          0   03/15/2017
6  510140   4256          0   07/25/2016
7  126320    110     375000   10/30/2017

In [146]:
len(sale_info)

266390

In [162]:
sale_info['SalePrice'].describe()

count    2.663900e+05
mean     5.273181e+05
std      1.099445e+06
min     -6.000000e+02
25%      0.000000e+00
50%      3.999500e+05
75%      7.000000e+05
max      6.250000e+07
Name: SalePrice, dtype: float64

In [148]:
sale_info['SalePrice'].value_counts()

0          83463
450000      1372
550000      1297
400000      1278
650000      1232
           ...  
320001         1
381479         1
1008241        1
565897         1
2430           1
Name: SalePrice, Length: 22774, dtype: int64

Given that there are 266390 entries in this data frame and 25% of the 'SalePrice' column is 0 (83463 entries), I've decided to drop the rows with no sale price, as replacing missing values with the column mean could greatly skew the data and adversely effect the results.

In [163]:
sale_info.drop(sale_info.loc[sale_info['SalePrice']<=1].index, inplace=True)
sale_info['SalePrice'].value_counts()

450000     1372
550000     1297
400000     1278
650000     1232
600000     1214
           ... 
464815        1
1560500       1
237502        1
868330        1
1851400       1
Name: SalePrice, Length: 22767, dtype: int64

In [164]:
len(sale_info)

182890

In [165]:
sale_info.head()

Major  Minor  SalePrice DocumentDate
1   638580    110     190000   07/14/2015
3   894677    240     818161   12/21/2016
7   126320    110     375000   10/30/2017
9   213043    120     560000   12/20/2019
10  940652    630     435000   07/22/2019

In [166]:
sale_info['SalePrice'].describe()

count    1.828900e+05
mean     7.680697e+05
std      1.255287e+06
min      2.000000e+00
25%      3.840000e+05
50%      5.750000e+05
75%      8.350000e+05
max      6.250000e+07
Name: SalePrice, dtype: float64

In [167]:
housing_data = pd.read_csv('../../data/raw/EXTR_ResBldg.csv')
housing_data.head(2)

Major  Minor  BldgNbr  NbrLivingUnits                     Address  \
0   9800    720        1               1  27719  SE 26TH WAY   98075   
1   9802    140        1               1  2829   277TH TER SE  98075   

  BuildingNumber Fraction DirectionPrefix                 StreetName  \
0          27719                       SE  26TH                        
1          2829                            277TH                       

  StreetType  ... FpMultiStory FpFreestanding  FpAdditional  YrBuilt  \
0     WAY     ...            0              0             0     2001   
1     TER     ...            0              0             0     2004   

   YrRenovated  PcntComplete  Obsolescence  PcntNetCondition  Condition  \
0            0             0             0                 0          3   
1            0             0             0                 0          3   

   AddnlCost  
0          0  
1          0  

[2 rows x 50 columns]

In [125]:
housing_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181510 entries, 0 to 181509
Data columns (total 50 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   Major               181510 non-null  int64  
 1   Minor               181510 non-null  int64  
 2   BldgNbr             181510 non-null  int64  
 3   NbrLivingUnits      181510 non-null  int64  
 4   Address             181510 non-null  object 
 5   BuildingNumber      181510 non-null  object 
 6   Fraction            181510 non-null  object 
 7   DirectionPrefix     181146 non-null  object 
 8   StreetName          181510 non-null  object 
 9   StreetType          181510 non-null  object 
 10  DirectionSuffix     181146 non-null  object 
 11  ZipCode             154594 non-null  object 
 12  Stories             181510 non-null  float64
 13  BldgGrade           181510 non-null  int64  
 14  BldgGradeVar        181510 non-null  int64  
 15  SqFt1stFloor        181510 non-nul

In [168]:
#For the sake of streamlining, I will merge the bath columns and combine number of bathrooms and create a column called 
#'Bathrooms' with the total amount of bathrooms (regardless of its size), per home.
housing_data['Bathrooms'] = housing_data[['BathFullCount', 'BathHalfCount', 'Bath3qtrCount']].astype(float).sum(1)
to_drop= ['BathFullCount', 'BathHalfCount', 'Bath3qtrCount']
housing_data.drop(to_drop, axis=1, inplace=True)
housing_data.head()

Major  Minor  BldgNbr  NbrLivingUnits                     Address  \
0   9800    720        1               1  27719  SE 26TH WAY   98075   
1   9802    140        1               1  2829   277TH TER SE  98075   
2   9830     20        1               1  1715   298TH CRESENT  SE     
3   9830    160        1               1  1861   297TH WAY SE  98024   
4  10050    180        1               1    35410   25TH PL S  98003   

  BuildingNumber Fraction DirectionPrefix                 StreetName  \
0          27719                       SE  26TH                        
1          2829                            277TH                       
2          1715                            298TH CRESENT               
3          1861                            297TH                       
4          35410                           25TH                        

  StreetType  ... FpFreestanding FpAdditional  YrBuilt  YrRenovated  \
0     WAY     ...              0            0     2001            0   
1     TER     ...              0            0     2004            0   
2             ...              0            0     2017            0   
3     WAY     ...              0            0     2013            0   
4     PL      ...              0            0     1994            0   

   PcntComplete  Obsolescence  PcntNetCondition  Condition  AddnlCost  \
0             0             0                 0          3          0   
1             0             0                 0          3          0   
2             0             0                 0          3          0   
3             0             0                 0          3          0   
4             0             0                 0          4          0   

   Bathrooms  
0        4.0  
1        5.0  
2        5.0  
3        4.0  
4        3.0  

[5 rows x 48 columns]

'saleprice','',,'waterfrontlocation''township''airportnoise','waterfrontfootage'

date price bedrooms bathrooms sqft_living sqft_lot floors condition grade sqft_above sqft_basement yr_built yr_renovated zipcode 

In [169]:
housing_data = housing_data[['Major', 'Minor', 'ZipCode','Stories','BldgGrade','Condition','Bedrooms','Bathrooms',
                             'SqFtTotLiving','SqFtTotBasement','SqFtOpenPorch','SqFtEnclosedPorch','SqFtDeck','HeatSystem',
                            'YrBuilt', 'YrRenovated']]
housing_data.head()

Major  Minor ZipCode  Stories  BldgGrade  Condition  Bedrooms  Bathrooms  \
0   9800    720   98075      2.0         11          3         4        4.0   
1   9802    140   98075      2.0         10          3         4        5.0   
2   9830     20     NaN      2.0         10          3         4        5.0   
3   9830    160   98024      2.0         10          3         4        4.0   
4  10050    180   98003      2.0          7          4         3        3.0   

   SqFtTotLiving  SqFtTotBasement  SqFtOpenPorch  SqFtEnclosedPorch  SqFtDeck  \
0           4100                0              0                  0         0   
1           3010                0            380                  0         0   
2           5080                0            360                  0       270   
3           4070                0            690                  0         0   
4           1610                0             60                  0         0   

   HeatSystem  YrBuilt  YrRenovated  
0           5     2001            0  
1           5     2004            0  
2           5     2017            0  
3           5     2013            0  
4           5     1994            0

In [170]:
housing_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181510 entries, 0 to 181509
Data columns (total 16 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Major              181510 non-null  int64  
 1   Minor              181510 non-null  int64  
 2   ZipCode            154594 non-null  object 
 3   Stories            181510 non-null  float64
 4   BldgGrade          181510 non-null  int64  
 5   Condition          181510 non-null  int64  
 6   Bedrooms           181510 non-null  int64  
 7   Bathrooms          181510 non-null  float64
 8   SqFtTotLiving      181510 non-null  int64  
 9   SqFtTotBasement    181510 non-null  int64  
 10  SqFtOpenPorch      181510 non-null  int64  
 11  SqFtEnclosedPorch  181510 non-null  int64  
 12  SqFtDeck           181510 non-null  int64  
 13  HeatSystem         181510 non-null  int64  
 14  YrBuilt            181510 non-null  int64  
 15  YrRenovated        181510 non-null  int64  
dtypes:

In [21]:
##**********NO LONGER NEED************
#Moving 'Bathrooms' column to just after 'Bedrooms'
bath = housing_data['Bathrooms']
housing_data.drop(labels=['Bathrooms'], axis=1, inplace = True)
housing_data.insert(18, 'Bathrooms', bath) #<-- '18' represents the column index number that I want to assign to this column.
#The column just after 'Bedrooms' is 'YrBuilt' whose index number is 18.
housing_data.head()

Major  Minor  Stories  BldgGrade  SqFt1stFloor  SqFt2ndFloor  \
0   9800    720      2.0         11          1970          2130   
1   9802    140      2.0         10          1610          1400   
2   9830     20      2.0         10          2520          2560   
3   9830    160      2.0         10          2210          1860   
4  10050    180      2.0          7           910           700   

   SqFtUpperFloor  SqFtTotLiving  SqFtTotBasement  SqFtFinBasement  ...  \
0               0           4100                0                0  ...   
1               0           3010                0                0  ...   
2               0           5080                0                0  ...   
3               0           4070                0                0  ...   
4               0           1610                0                0  ...   

   SqFtGarageAttached  SqFtOpenPorch  SqFtEnclosedPorch  SqFtDeck  HeatSystem  \
0                 750              0                  0         0           5   
1                 660            380                  0         0           5   
2                1020            360                  0       270           5   
3                1000            690                  0         0           5   
4                 440             60                  0         0           5   

   HeatSource  Bedrooms  Bathrooms  YrBuilt  Condition  
0           2         4        4.0     2001          3  
1           2         4        5.0     2004          3  
2           2         4        5.0     2017          3  
3           2         4        4.0     2013          3  
4           2         3        3.0     1994          4  

[5 rows x 21 columns]

In [171]:
parcel = pd.read_csv('../../data/raw/EXTR_Parcel.csv')
parcel.head(2)

Unnamed: 0   Major  Minor PropName                            PlatName  \
0           0  807841    410                      SUMMER RIDGE DIV NO. 02   
1           2  755080     15           SANDER'S TO GILMAN PK & SALMON BAY   

          PlatLot PlatBlock  Range  Township  Section  ... SeismicHazard  \
0  41                            6        25       22  ...             N   
1  3                1            3        25       11  ...             N   

  LandslideHazard  SteepSlopeHazard  Stream  Wetland  SpeciesOfConcern  \
0               N                 N       N        N                 N   
1               N                 N       N        N                 N   

  SensitiveAreaTract  WaterProblems TranspConcurrency  OtherProblems  
0                  N              N                 N              N  
1                  N              N                 N              N  

[2 rows x 82 columns]

In [31]:
parcel['WfntLocation'].value_counts()

0    200463
8      1201
3      1037
6      1018
9       981
7       346
1        53
5        51
4        41
2         8
Name: WfntLocation, dtype: int64

In [23]:
parcel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 205199 entries, 0 to 205198
Data columns (total 82 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Unnamed: 0              205199 non-null  int64  
 1   Major                   205199 non-null  int64  
 2   Minor                   205199 non-null  int64  
 3   PropName                196088 non-null  object 
 4   PlatName                176654 non-null  object 
 5   PlatLot                 205199 non-null  object 
 6   PlatBlock               205199 non-null  object 
 7   Range                   205199 non-null  int64  
 8   Township                205199 non-null  int64  
 9   Section                 205199 non-null  int64  
 10  QuarterSection          205199 non-null  object 
 11  PropType                205199 non-null  object 
 12  Area                    205193 non-null  float64
 13  SubArea                 205193 non-null  float64
 14  SpecArea            

In [ ]:
'saleprice','',,'waterfrontlocation''township''airportnoise','waterfrontfootage'

In [172]:
parcel = parcel[['Major', 'Minor', 'SqFtLot','Township','AirportNoise', 'WfntFootage']]
parcel.head()

Major  Minor  SqFtLot  Township  AirportNoise  WfntFootage
0  807841    410     7424        25             0            0
1  755080     15     5000        25             0            0
2  888600    135   277041        22             0            0
3   22603   9181    10560        26             0            0
4  229670    160     9853        26             0            0

In [58]:
display(parcel['TrafficNoise'].value_counts())
display("Traffic noise ranges on a scale of 0 to 3, 0 is no noise, 1 being moderate, 2 being high, and 3 being extreme")

0    176076
1     16736
2     10563
3      1824
Name: TrafficNoise, dtype: int64

'Traffic noise ranges on a scale of 0 to 3, 0 is no noise, 1 being moderate, 2 being high, and 3 being extreme'

In [59]:
parcel['OtherNuisances'].value_counts()

N    199057
Y      6142
Name: OtherNuisances, dtype: int64

In [60]:
#Replacing 'N' with 0 and 'Y' with 1 (No/Yes) for easy plotting
parcel['OtherNuisances']=parcel['OtherNuisances'].replace('N',0)
parcel['OtherNuisances']=parcel['OtherNuisances'].replace('Y',1)
parcel['OtherNuisances'].value_counts()

0    199057
1      6142
Name: OtherNuisances, dtype: int64

I'll now merge my data frames for one clean file:

In [173]:
house_data = pd.merge(sale_info, housing_data,  how='left', left_on=['Major','Minor'], right_on = ['Major','Minor'])
house_data.head()

Major  Minor  SalePrice DocumentDate ZipCode  Stories  BldgGrade  \
0  638580    110     190000   07/14/2015   98188      1.0        7.0   
1  894677    240     818161   12/21/2016     NaN      2.0        9.0   
2  126320    110     375000   10/30/2017   98106      2.0        6.0   
3  213043    120     560000   12/20/2019   98019      1.0        7.0   
4  940652    630     435000   07/22/2019   98038      2.0        7.0   

   Condition  Bedrooms  Bathrooms  SqFtTotLiving  SqFtTotBasement  \
0        3.0       4.0        2.0         2130.0           1050.0   
1        3.0       4.0        3.0         2450.0              0.0   
2        3.0       3.0        1.0         1650.0              0.0   
3        3.0       3.0        2.0         1930.0              0.0   
4        3.0       3.0        3.0         1975.0              0.0   

   SqFtOpenPorch  SqFtEnclosedPorch  SqFtDeck  HeatSystem  YrBuilt  \
0            0.0                0.0     300.0         5.0   1963.0   
1           40.0                0.0      60.0         5.0   2016.0   
2            0.0                0.0       0.0         4.0   1919.0   
3          180.0                0.0       0.0         5.0   1989.0   
4           45.0                0.0       0.0         5.0   1994.0   

   YrRenovated  
0          0.0  
1          0.0  
2          0.0  
3          0.0  
4          0.0

In [130]:
house_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 185082 entries, 0 to 185081
Data columns (total 16 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Major              185082 non-null  int64  
 1   Minor              185082 non-null  int64  
 2   SalePrice          185082 non-null  int64  
 3   DocumentDate       185082 non-null  object 
 4   ZipCode            141551 non-null  object 
 5   Stories            173128 non-null  float64
 6   BldgGrade          173128 non-null  float64
 7   Condition          173128 non-null  float64
 8   Bedrooms           173128 non-null  float64
 9   Bathrooms          173128 non-null  float64
 10  SqFtTotLiving      173128 non-null  float64
 11  SqFtTotBasement    173128 non-null  float64
 12  SqFtOpenPorch      173128 non-null  float64
 13  SqFtEnclosedPorch  173128 non-null  float64
 14  SqFtDeck           173128 non-null  float64
 15  HeatSystem         173128 non-null  float64
dtypes:

In [174]:
house_data = pd.merge(house_data, parcel,  how='left', left_on=['Major','Minor'], right_on = ['Major','Minor'])
house_data.head()

Major  Minor  SalePrice DocumentDate ZipCode  Stories  BldgGrade  \
0  638580    110     190000   07/14/2015   98188      1.0        7.0   
1  894677    240     818161   12/21/2016     NaN      2.0        9.0   
2  126320    110     375000   10/30/2017   98106      2.0        6.0   
3  213043    120     560000   12/20/2019   98019      1.0        7.0   
4  940652    630     435000   07/22/2019   98038      2.0        7.0   

   Condition  Bedrooms  Bathrooms  ...  SqFtOpenPorch  SqFtEnclosedPorch  \
0        3.0       4.0        2.0  ...            0.0                0.0   
1        3.0       4.0        3.0  ...           40.0                0.0   
2        3.0       3.0        1.0  ...            0.0                0.0   
3        3.0       3.0        2.0  ...          180.0                0.0   
4        3.0       3.0        3.0  ...           45.0                0.0   

   SqFtDeck  HeatSystem  YrBuilt  YrRenovated  SqFtLot  Township  \
0     300.0         5.0   1963.0          0.0  10534.0      23.0   
1      60.0         5.0   2016.0          0.0   3813.0      26.0   
2       0.0         4.0   1919.0          0.0   4760.0      24.0   
3       0.0         5.0   1989.0          0.0  10523.0      26.0   
4       0.0         5.0   1994.0          0.0   7392.0      22.0   

   AirportNoise  WfntFootage  
0           0.0          0.0  
1           0.0          0.0  
2           0.0          0.0  
3           0.0          0.0  
4           0.0          0.0  

[5 rows x 22 columns]

In [133]:
house_data.describe()

Major          Minor     SalePrice        Stories  \
count  185082.000000  185082.000000  1.850820e+05  173128.000000   
mean   444588.033061    1706.162847  7.706113e+05       1.552063   
std    287962.954145    2995.361836  1.265662e+06       0.541694   
min       100.000000       1.000000  1.000000e+00       1.000000   
25%    198020.000000     120.000000  3.840000e+05       1.000000   
50%    375160.000000     340.000000  5.750000e+05       1.500000   
75%    722850.000000    1140.000000  8.350000e+05       2.000000   
max    990600.000000    9694.000000  6.250000e+07       4.000000   

           BldgGrade      Condition       Bedrooms      Bathrooms  \
count  173128.000000  173128.000000  173128.000000  173128.000000   
mean        7.733365       3.464090       3.460538       2.640116   
std         1.199443       0.709053       0.975715       1.045961   
min         1.000000       1.000000       0.000000       0.000000   
25%         7.000000       3.000000       3.000000       2.000000   
50%         8.000000       3.000000       3.000000       3.000000   
75%         8.000000       4.000000       4.000000       3.000000   
max        13.000000       5.000000      24.000000      16.000000   

       SqFtTotLiving  SqFtTotBasement  SqFtOpenPorch  SqFtEnclosedPorch  \
count  173128.000000    173128.000000  173128.000000      173128.000000   
mean     2194.646302       469.046907      79.311966           5.858469   
std      1009.484465       582.078240     133.937209          39.415613   
min         1.000000         0.000000       0.000000           0.000000   
25%      1470.000000         0.000000       0.000000           0.000000   
50%      2000.000000         0.000000      40.000000           0.000000   
75%      2730.000000       940.000000     110.000000           0.000000   
max     14980.000000      5610.000000   11570.000000        2330.000000   

            SqFtDeck     HeatSystem       SqFtLot       Township  \
count  173128.000000  173128.000000  1.839100e+05  183910.000000   
mean      135.365400       4.970115  4.346036e+04      23.756055   
std       399.453401       1.076105  7.090724e+05       1.673932   
min         0.000000       0.000000  0.000000e+00       0.000000   
25%         0.000000       5.000000  5.000000e+03      22.000000   
50%         0.000000       5.000000  7.592000e+03      24.000000   
75%       210.000000       5.000000  1.120000e+04      25.000000   
max    140000.000000       8.000000  2.330525e+07      26.000000   

       StreetSurface  InadequateParking   TrafficNoise  OtherNuisances  
count  183910.000000      183910.000000  183910.000000   183910.000000  
mean        1.017650           1.263814       0.221826        0.031771  
std         0.248127           0.963883       0.578674        0.175390  
min         0.000000           0.000000       0.000000        0.000000  
25%         1.000000           0.000000       0.000000        0.000000  
50%         1.000000           2.000000       0.000000        0.000000  
75%         1.000000           2.000000       0.000000        0.000000  
max         4.000000           2.000000       3.000000        1.000000

In [134]:
house_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 185082 entries, 0 to 185081
Data columns (total 23 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Major              185082 non-null  int64  
 1   Minor              185082 non-null  int64  
 2   SalePrice          185082 non-null  int64  
 3   DocumentDate       185082 non-null  object 
 4   ZipCode            141551 non-null  object 
 5   Stories            173128 non-null  float64
 6   BldgGrade          173128 non-null  float64
 7   Condition          173128 non-null  float64
 8   Bedrooms           173128 non-null  float64
 9   Bathrooms          173128 non-null  float64
 10  SqFtTotLiving      173128 non-null  float64
 11  SqFtTotBasement    173128 non-null  float64
 12  SqFtOpenPorch      173128 non-null  float64
 13  SqFtEnclosedPorch  173128 non-null  float64
 14  SqFtDeck           173128 non-null  float64
 15  HeatSystem         173128 non-null  float64
 16  Sq

In [175]:
#Checking for any missing data:
for column in list(house_data.columns):
    print(column, sum(house_data[column].isnull()))

Major 0
Minor 0
SalePrice 0
DocumentDate 0
ZipCode 43513
Stories 11937
BldgGrade 11937
Condition 11937
Bedrooms 11937
Bathrooms 11937
SqFtTotLiving 11937
SqFtTotBasement 11937
SqFtOpenPorch 11937
SqFtEnclosedPorch 11937
SqFtDeck 11937
HeatSystem 11937
YrBuilt 11937
YrRenovated 11937
SqFtLot 1157
Township 1157
AirportNoise 1157
WfntFootage 1157


A number of the columns have missing data and to attempt to replace the null values with the median is not feasible (the sqft of a lot varies greatly and can't be averaged out), so I will rows containing empty values:

In [176]:
house_data = house_data.dropna(how='any',axis=0)
house_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 141548 entries, 0 to 185059
Data columns (total 22 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Major              141548 non-null  int64  
 1   Minor              141548 non-null  int64  
 2   SalePrice          141548 non-null  int64  
 3   DocumentDate       141548 non-null  object 
 4   ZipCode            141548 non-null  object 
 5   Stories            141548 non-null  float64
 6   BldgGrade          141548 non-null  float64
 7   Condition          141548 non-null  float64
 8   Bedrooms           141548 non-null  float64
 9   Bathrooms          141548 non-null  float64
 10  SqFtTotLiving      141548 non-null  float64
 11  SqFtTotBasement    141548 non-null  float64
 12  SqFtOpenPorch      141548 non-null  float64
 13  SqFtEnclosedPorch  141548 non-null  float64
 14  SqFtDeck           141548 non-null  float64
 15  HeatSystem         141548 non-null  float64
 16  Yr

In [137]:
#Rechecking for any missing data:
for column in list(house_data.columns):
    print(column, sum(house_data[column].isnull()))

Major 0
Minor 0
SalePrice 0
DocumentDate 0
ZipCode 0
Stories 0
BldgGrade 0
Condition 0
Bedrooms 0
Bathrooms 0
SqFtTotLiving 0
SqFtTotBasement 0
SqFtOpenPorch 0
SqFtEnclosedPorch 0
SqFtDeck 0
HeatSystem 0
SqFtLot 0
Township 0
DistrictName 0
StreetSurface 0
InadequateParking 0
TrafficNoise 0
OtherNuisances 0


In [138]:
len(house_data)

141551

In [139]:
house_data.tail(2)

Major  Minor  SalePrice DocumentDate ZipCode  Stories  BldgGrade  \
185079  886030    550     900000   07/16/2016   98052      1.0        8.0   
185080  769791     30     133000   07/16/2016   98030      2.0        7.0   

        Condition  Bedrooms  Bathrooms  ...  SqFtEnclosedPorch  SqFtDeck  \
185079        3.0       4.0        3.0  ...                0.0     630.0   
185080        3.0       3.0        3.0  ...                0.0       0.0   

        HeatSystem  SqFtLot  Township  DistrictName  StreetSurface  \
185079         5.0  13238.0      25.0       REDMOND            1.0   
185080         5.0   5936.0      22.0          KENT            1.0   

        InadequateParking TrafficNoise  OtherNuisances  
185079                2.0          0.0             0.0  
185080                0.0          0.0             0.0  

[2 rows x 23 columns]

In [177]:
#Resetting the index
house_data.reset_index(drop=True, inplace=True)

In [178]:
house_data.tail(2)

Major  Minor  SalePrice DocumentDate ZipCode  Stories  BldgGrade  \
141546  886030    550     900000   07/16/2016   98052      1.0        8.0   
141547  769791     30     133000   07/16/2016   98030      2.0        7.0   

        Condition  Bedrooms  Bathrooms  ...  SqFtOpenPorch  SqFtEnclosedPorch  \
141546        3.0       4.0        3.0  ...            0.0                0.0   
141547        3.0       3.0        3.0  ...           40.0                0.0   

        SqFtDeck  HeatSystem  YrBuilt  YrRenovated  SqFtLot  Township  \
141546     630.0         5.0   1975.0          0.0  13238.0      25.0   
141547       0.0         5.0   1990.0          0.0   5936.0      22.0   

        AirportNoise  WfntFootage  
141546           0.0          0.0  
141547           0.0          0.0  

[2 rows x 22 columns]

### Converting the 'ZipCode' Column to An (int) For Plotting

In [180]:
house_data['ZipCode'] = house_data['ZipCode'].str.split('-').str[0]
house_data['ZipCode'].value_counts()

98042    4432
98023    4037
98038    3809
98058    3622
98115    3476
         ... 
98012       1
98043       1
95059       1
91844       1
98104       1
Name: ZipCode, Length: 115, dtype: int64

In [182]:
house_data['ZipCode'].describe()

count     141548
unique       115
top        98042
freq        4432
Name: ZipCode, dtype: object

In [183]:
house_data['ZipCode'] = house_data['ZipCode'].astype(str).str.replace(',', '')
house_data['ZipCode'] = house_data['ZipCode'].astype(str).str.replace(' ', '')
house_data['ZipCode'] = house_data['ZipCode'].astype(str).str.replace('.', '')

In [185]:
df = house_data[house_data['ZipCode'].apply(lambda x: not x.isnumeric())]
df

Major  Minor  SalePrice DocumentDate   ZipCode  Stories  BldgGrade  \
24179  982870   2305    1330000   12/29/2015  Aonplans      3.0       10.0   
26953  312606   9112     322000   04/30/2018                1.0        7.0   
67461  982870   2305    1900000   02/22/2016  Aonplans      3.0       10.0   

       Condition  Bedrooms  Bathrooms  ...  SqFtOpenPorch  SqFtEnclosedPorch  \
24179        3.0       3.0        4.0  ...            0.0               70.0   
26953        3.0       2.0        1.0  ...          100.0                0.0   
67461        3.0       3.0        4.0  ...            0.0               70.0   

       SqFtDeck  HeatSystem  YrBuilt  YrRenovated  SqFtLot  Township  \
24179     600.0         7.0   2019.0          0.0   1424.0      25.0   
26953       0.0         5.0   2015.0          0.0  77536.0      26.0   
67461     600.0         7.0   2019.0          0.0   1424.0      25.0   

       AirportNoise  WfntFootage  
24179           0.0          0.0  
26953           0.0          0.0  
67461           0.0          0.0  

[3 rows x 22 columns]

In [186]:
house_data = house_data.drop(index=[24179,26953,67461])
house_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 141545 entries, 0 to 141547
Data columns (total 22 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Major              141545 non-null  int64  
 1   Minor              141545 non-null  int64  
 2   SalePrice          141545 non-null  int64  
 3   DocumentDate       141545 non-null  object 
 4   ZipCode            141545 non-null  object 
 5   Stories            141545 non-null  float64
 6   BldgGrade          141545 non-null  float64
 7   Condition          141545 non-null  float64
 8   Bedrooms           141545 non-null  float64
 9   Bathrooms          141545 non-null  float64
 10  SqFtTotLiving      141545 non-null  float64
 11  SqFtTotBasement    141545 non-null  float64
 12  SqFtOpenPorch      141545 non-null  float64
 13  SqFtEnclosedPorch  141545 non-null  float64
 14  SqFtDeck           141545 non-null  float64
 15  HeatSystem         141545 non-null  float64
 16  Yr

In [187]:
house_data['ZipCode']= house_data['ZipCode'].astype(np.int64)

In [188]:
house_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 141545 entries, 0 to 141547
Data columns (total 22 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Major              141545 non-null  int64  
 1   Minor              141545 non-null  int64  
 2   SalePrice          141545 non-null  int64  
 3   DocumentDate       141545 non-null  object 
 4   ZipCode            141545 non-null  int64  
 5   Stories            141545 non-null  float64
 6   BldgGrade          141545 non-null  float64
 7   Condition          141545 non-null  float64
 8   Bedrooms           141545 non-null  float64
 9   Bathrooms          141545 non-null  float64
 10  SqFtTotLiving      141545 non-null  float64
 11  SqFtTotBasement    141545 non-null  float64
 12  SqFtOpenPorch      141545 non-null  float64
 13  SqFtEnclosedPorch  141545 non-null  float64
 14  SqFtDeck           141545 non-null  float64
 15  HeatSystem         141545 non-null  float64
 16  Yr

In [189]:
#Saving as a .csv:
data = house_data.to_csv('data.csv', index = True) 

In [193]:
data = pd.read_csv('../../data/processed/data.csv', index_col=0) #<-Load w/out the unnecessary 'Unnamed:0' column.

In [194]:
data.describe()

Major          Minor     SalePrice       ZipCode  \
count  141545.000000  141545.000000  1.415450e+05  1.415450e+05   
mean   448728.218736    1562.838405  6.942912e+05  1.050155e+05   
std    286620.447904    2851.067000  8.054475e+05  2.607710e+06   
min       100.000000       1.000000  5.000000e+00  3.000000e+00   
25%    202504.000000     115.000000  3.790000e+05  9.803200e+04   
50%    383206.000000     326.000000  5.560000e+05  9.805900e+04   
75%    722921.000000    1017.000000  8.000000e+05  9.811800e+04   
max    990600.000000    9689.000000  6.250000e+07  9.811820e+08   

             Stories      BldgGrade      Condition       Bedrooms  \
count  141545.000000  141545.000000  141545.000000  141545.000000   
mean        1.440351       7.577541       3.565227       3.402049   
std         0.507238       1.190614       0.744951       0.950776   
min         1.000000       1.000000       1.000000       0.000000   
25%         1.000000       7.000000       3.000000       3.000000   
50%         1.000000       7.000000       3.000000       3.000000   
75%         2.000000       8.000000       4.000000       4.000000   
max         4.000000      13.000000       5.000000      15.000000   

           Bathrooms  SqFtTotLiving  ...  SqFtOpenPorch  SqFtEnclosedPorch  \
count  141545.000000  141545.000000  ...  141545.000000      141545.000000   
mean        2.495871    2089.849596  ...      65.869031           6.481691   
std         1.006218     956.901555  ...     121.868734          42.291230   
min         0.000000       2.000000  ...       0.000000           0.000000   
25%         2.000000    1420.000000  ...       0.000000           0.000000   
50%         3.000000    1910.000000  ...       0.000000           0.000000   
75%         3.000000    2550.000000  ...      90.000000           0.000000   
max        16.000000   14980.000000  ...   11570.000000        2330.000000   

            SqFtDeck     HeatSystem        YrBuilt    YrRenovated  \
count  141545.000000  141545.000000  141545.000000  141545.000000   
mean      140.719750       4.912925    1969.382493     116.526278   
std       218.450465       1.044588      29.042281     468.617484   
min         0.000000       0.000000    1900.000000       0.000000   
25%         0.000000       5.000000    1951.000000       0.000000   
50%         0.000000       5.000000    1970.000000       0.000000   
75%       220.000000       5.000000    1992.000000       0.000000   
max      4290.000000       8.000000    2020.000000    2020.000000   

            SqFtLot       Township  AirportNoise    WfntFootage  
count  1.415450e+05  141545.000000      141545.0  141545.000000  
mean   4.588788e+04      23.792412           0.0       8.071666  
std    8.046268e+05       1.650141           0.0     183.441218  
min    3.750000e+02      19.000000           0.0       0.000000  
25%    5.500000e+03      23.000000           0.0       0.000000  
50%    7.980000e+03      24.000000           0.0       0.000000  
75%    1.140000e+04      25.000000           0.0       0.000000  
max    2.330525e+07      26.000000           0.0    5300.000000  

[8 rows x 21 columns]